### ISSUE TITLE
Muttator doen't work with Thunderbird 38.0.1 on Linux

### ISSUE THREAD

**Issue by kevincox** [vimperator/vimperator-labs#237]
None of the keybindings work and there is nothing in the status bar. Clicking on the preferences brings up an empty window.  There are no muttator related messages in the console.

**Comment by stevleibelt** [comment]
Same for me :-(

**Comment by w1ld** [comment]
Looks like I have the same on Thunderbird 38.0.1, Mutator 1.2, Windows. Mutator does not function: no status bar, no keys working. Errors:

	Timestamp: 19.06.2015 10:41:00
	Error: Error opening input stream (invalid filename?): liberator://template/chrome://liberator/content/mail.js
	Source File: chrome://liberator/content/liberator-overlay.js
	Line: 24
	
	Timestamp: 19.06.2015 10:41:00
	Error: [Exception... "Component returned failure code: 0x80520012 (NS_ERROR_FILE_NOT_FOUND) [nsIChannel.open]"  nsresult: "0x80520012 (NS_ERROR_FILE_NOT_FOUND)"  location: "JS frame :: jar:file:///C:/Users/wild/AppData/Roaming/Thunderbird/Profiles/qbyfwg2v.default/extensions/muttator@mozdev.org.xpi!/components/protocols.js :: Liberator.prototype.newChannel :: line 233"  data: no]
	Source File: jar:file:///C:/Users/wild/AppData/Roaming/Thunderbird/Profiles/qbyfwg2v.default/extensions/muttator@mozdev.org.xpi!/components/protocols.js
	Line: 249
	
	Timestamp: 19.06.2015 10:41:00
	Error: plugins is not defined
	Source File: chrome://liberator/content/liberator-overlay.js -> liberator://template/chrome://liberator/content/io.js
	Line: 10
	
	Timestamp: 19.06.2015 10:41:00
	Error: redeclaration of let usage
	Source File: chrome://liberator/content/liberator-overlay.js -> liberator://template/chrome://liberator/content/liberator.js
	Line: 1720, Column: 24
	Source Code:
	                    var usage = template.genericOutput(config.name + " Usage", usage
	
	
	Timestamp: 19.06.2015 10:41:00
	Error: redeclaration of var str
	Source File: chrome://liberator/content/liberator-overlay.js -> liberator://template/chrome://liberator/content/commandline.js
	Line: 1469, Column: 24
	Source Code:
	                    let str = CommandLine.echoArgumentToString(str, true);

**Comment by cvmocanu** [comment]
Same for me on Windows 8.1. It started happening since I upgraded to Thunderbird 38.0.1. Without muttator it feels like a big downgrade...

**Comment by stevleibelt** [comment]
Indeed, it is like having a not working tool (don't worry, I already donated to vimperator labs a week ago :-)).

**Comment by maxauthority** [comment]
Has anyone tried to build the latest version from git? Someone who did a
review managed to get Muttator working on TB38:

In order to get muttator to work with Thunderbird 38, do the following (in
any linux shell; you need to have git and possibly other things installed):

`git clone https://github.com/vimperator/vimperator-labs.git`

Go to muttator dir and do `make xpi`.

You will find a working xpi in the downloads folder.

-----
If that works, I'll release a new version to the public.

**Comment by kevincox** [comment]
Yeah, I rebuilt from 737ada6a178480793efcfcbb4a48bb291c70d67c and it fixed it.

**Comment by hugoArregui** [comment]
Confirmed here too, building from git fixes the problem

**Comment by stevleibelt** [comment]
@maxauthority 

could you release a new version for addons.mozilla.org? Then the autoupdate could do its job and I don't have to maintain this on my own :-).

**Comment by maxauthority** [comment]
Yeah, I planned do do that. Will do it know, so I don't forget again :)

Thanks for the reminder!

**Comment by stevleibelt** [comment]
@maxauthority 
*remind*, still version 1.2 on the official [addons](https://addons.mozilla.org/en-US/thunderbird/addon/muttator) page.
 :)

**Comment by stevleibelt** [comment]
reminder, still version 1.2 on the official [addons](https://addons.mozilla.org/en-US/thunderbird/addon/muttator/) page.
:-)

**Comment by kevincox** [comment]
This just started happening on my other computer...it would be nice to have an update.

**Comment by Shade30** [comment]
Just a kind reminder to update a version at addons.mozilla.org.

**Comment by RonaldDuncan** [comment]
Still not there :)

**Comment by maxauthority** [comment]
Yeah, I uploaded a version a week ago or so, but it was rejected because of:
---------

Your add-on, Muttator 1.3, has been reviewed by an editor and did not meet
the criteria for being hosted in our gallery.


Reviewer:
Leszek Życzkowski

Comments:
Your version was rejected because of the following problems:

1. common/content/liberator.xul:9
JavaScript Compile-Time Error. A compile-time error in the JavaScript
halted validation of that file.Illegal character.

Please fix them and submit again. Thank you.
-------

I have no clue why, as this file shouldn't have changed in a while :(

Can somebody try to do build an XPI of the current git and test it with a
recent Thunderbird?

**Comment by gkatsev** [comment]
@maxauthority I wonder if any of the fixes I have in master now would help.

**Comment by RonaldDuncan** [comment]
I just built it and it worked fine for me on TB 38.2.0 (Linux Ubuntu).

**Comment by RonaldDuncan** [comment]
line 9 is blank with just a line feed following the previous LF which is previously repeated earlier so I can no see any reason for the compile error unless something weird happened for the reviewer compiling on a mac/windows??

It has not been changed for a while.

**Comment by gkatsev** [comment]
It's possible that while it complains about line 9 of the xul file, the error is actually elsewhere.

**Comment by RonaldDuncan** [comment]
TB 38.2.0 on Ubuntu 14.04.3 LTS

**Comment by morsedl** [comment]
I re-installed thunderbird, thunderbird-gnome-support, and thunderbird-globalmenu (v38.2, from Ubuntu 12.04 repos) and uninstalled muttator.  Then I ran "git clone https://github.com/vimperator/vimperator-labs.git", changed to the muttator subdirectory, and ran "make xpi" without errors.  I then installed the resulting .xpi file in ../downloads into Thunderbird.

Unfortunately, no joy: None of the keyboard input seems to be getting to Muttator.  So, I am again back to stock Thunderbird v31.8 (from bz2 file from the Mozilla site directly).

Do I need to pull commit 737ada6 specifically, as @kevincox did?

**Comment by mbr0wn** [comment]
I just pulled HEAD today, and it worked. On TB 38.3.0.

**Comment by morsedl** [comment]
Thank you posting and letting us know mbr0wn!  I just re-installed 
thunderbird, thunderbird-gnome-support, thunderbird-globalmenu, and 
thunderbird-locale-en-us, and then did a 'git clone', ran 'make' in the 
muttator subdirectory, which created an muttator-1.3.xpi file in 
../downloads, which installed fine and I am back in business.  Again, 
thank you.  It's nice to have global menu support again, and not have 
thunderbird posting its own notification but rather using Unity again.

**Comment by maxauthority** [comment]
OK I'll try again to submit it to addons.mozilla.org

**Comment by amosbird** [comment]
It is still not working. I'm getting hash issues.
```
WARN	Download of https://addons.mozilla.org/thunderbird/downloads/file/370889/muttator-1.3.1-tb.xpi?src=api failed: Downloaded file hash (b1dd1d3d9def09da763136c69282a5e0f45c6e6fd950c76cfc4333b7cbe60c23) did not match provided hash (3c1a682bb0b22f3d845efa6ac0574f322599a216f15f0fffb2aee43c1914cea1)
```

**Comment by joelostblom** [comment]
I'm having the same problem with Muttator 1.3.1.1-let-fixed. I can't download the file and receive this error

```
1458357645733   addons.xpi      WARN    Download of https://addons.mozilla.org/thunderbird/downloads/file/370889/muttator-1.3.1-tb.xpi?src=api failed: Downloaded file hash (b1dd1d3d9def09da763136c69282a5e0f45c6e6fd950c76cfc4333b7cbe60c23) did not match provided hash (3c1a682bb0b22f3d845efa6ac0574f322599a216f15f0fffb2aee43c1914cea1)
```

Can this be fixed or should I download from github? If I download from github, will it still be automatically updated when new releases roll out?

**Comment by timss** [comment]
@joelostblom It should be possible to do the same with Thunderbird/Muttator as described here with Firefox/Vimperator: [CONTRIBUTING.md#development-installation](https://github.com/vimperator/vimperator-labs/blob/master/CONTRIBUTING.md#development-installation). Whether it'll resolve your issue I can't say unfortunately.

**Comment by antonmry** [comment]
I've just downloaded master and compile it for Thunderbird 45 on Fedora 23 and it works as usual. Muttator is an awesome plugin, thanks to the maintainers!

**Comment by gineadai** [comment]
I installed Thunderbird 45 and on startup error appear - `TyperError: ervices.get(...)defaultServer is null` - press ESC and gone, but error `TypeError: gDBView is null` in constantly in commandline. How to fix this?
